core/app/models/spree/order_inventory.rb

```diff
@@ -57,7 +57,7 @@ def remove(quantity, shipment = nil)
     # first unshipped that's leaving from a stock_location that stocks this variant, with availability check
     # first unshipped that's leaving from a stock_location that stocks this variant, without availability check
     def determine_target_shipment(quantity)
-      potential_shipments = order.shipments.select(&:ready_or_pending?)
+      potential_shipments = order.shipments.order(:created_at, :id).select(&:ready_or_pending?)
```

Comment:
I am a bit worried about how this can change the shipment used at the end of the checkout, when there are multiple available. Can you please help me understand why did you pick specifically this order?

Reply:
Assuming the spec is describing the desired behavior, the issue was clearly reproducible when reversing the order of creation. I think most databases will fetch records in order of creation when no `ORDER BY` clause is provided, but not all of them (see [this answer](https://stackoverflow.com/a/60582276) for example).

By explicitly requesting the creation order fixed the spec, and reversing it consistently made it fail.

The order is also disambiguated by the adding `:id` for when, like in tests, the `:created_at` time of two records can be the same because of lower date-time precision on the database.